### PR TITLE
issue #902

### LINKED ISSUE #902: Remove .Int64 calls from all sdk.Int instances.
**Is your feature request related to a problem? Please describe.**
Currently we could have an overflow whenever converting `sdk.Int` to int64. 

**Describe the solution you'd like**
Remove all int64 calls and replace them by `sdk.Int`.

**Additional Context**
Make sure all calls that return a `sdk.Int` do it in a string with decimal representation (currently it returns `1000000n`).

### PR DESCRIPTION
**Pull request by oten91**
Closes #902

### CODE REVIEW

File: x/pocketcore/types/proof.go [hunk 1]
```diff
@@ -210,20 +209,20 @@ type ChallengeProofInvalidData struct {
...
-func (c ChallengeProofInvalidData) ValidateLocal(h SessionHeader, maxRelays int64, supportedBlockchains []string, sessionNodeCount int, sessionNodes SessionNodes, selfAddr sdk.Address) sdk.Error {
+func (c ChallengeProofInvalidData) ValidateLocal(h SessionHeader, maxRelays sdk.Int, supportedBlockchains []string, sessionNodeCount int, sessionNodes SessionNodes, selfAddr sdk.Address) sdk.Error {
 	// check if verifyPubKey in session (must be in session to do challenges)
 	if !sessionNodes.ContainsAddress(selfAddr) {
 		return NewNodeNotInSessionError(ModuleName)
 	}
 	sessionblockHeight := h.SessionBlockHeight
 	// calculate the maximum possible challenges
-	maxPossibleChallenges := int64(math.Ceil(float64(maxRelays)/float64(len(supportedBlockchains))) / (float64(sessionNodeCount)))
+	maxPossibleChallenges := maxRelays.ToDec().Quo(sdk.NewDec(int64(len(supportedBlockchains)))).Quo(sdk.NewDec(int64(sessionNodeCount))).RoundInt()
```
Comment (oten91) [inline x/pocketcore/types/proof.go:219]: *

File: x/pocketcore/types/session.go [hunk 2]
```diff
@@ -272,6 +271,9 @@ func BlockHash(ctx sdk.Context) string {
 	return hex.EncodeToString(ctx.BlockHeader().LastBlockId.Hash)
 }
 
-func MaxPossibleRelays(app appexported.ApplicationI, sessionNodeCount int64) int64 {
-	return int64(math.Ceil(float64(app.GetMaxRelays().Int64())/float64(len(app.GetChains()))) / (float64(sessionNodeCount)))
+// "MaxPossibleRelays" - Returns the maximum possible amount of relays for an App on a sessions
+func MaxPossibleRelays(app appexported.ApplicationI, sessionNodeCount int64) sdk.Int {
+	//GetMaxRelays Max value is bound to math.MaxUint64,
+	//current worse case is 1 chain and 5 nodes per session with a result of 3689348814741910323 which can be used safely as int64
+	return app.GetMaxRelays().ToDec().Quo(sdk.NewDec(int64(len(app.GetChains())))).Quo(sdk.NewDec(sessionNodeCount)).RoundInt()
```
Comment (oten91) [inline x/pocketcore/types/session.go:278]: *

File: x/pocketcore/types/proof.go (hunk 1 above)
Comment (oten91) [reply x/pocketcore/types/proof.go:219]: Review Change on Calculation

File: x/pocketcore/types/session.go (hunk 2 above)
Comment (oten91) [reply x/pocketcore/types/session.go:278]: Review Change on Calculation

**Review by edjroz — APPROVED** [review]

**Pull request merged**